### ISSUE TITLE
Some confusion about the code

### ISSUE THREAD

**Issue by qilong-zhang** [dongyp13/Non-Targeted-Adversarial-Attacks#12]
```python
  noise = tf.gradients(cross_entropy, x)[0]
  noise = noise / tf.reduce_mean(tf.abs(noise), [1,2,3], keep_dims=True)
  noise = momentum * grad + noise
  x = x + alpha * tf.sign(noise)
  x = tf.clip_by_value(x, x_min, x_max)
  i = tf.add(i, 1)
```
I'm confused about this line
```python
  noise = noise / tf.reduce_mean(tf.abs(noise), [1,2,3], keep_dims=True)
```
L1 norm should be tf.reduce_sum()?

**Comment by dongyp13** [comment]
Right. We actually normalize the gradient according to each channel. Note that tf.reduce_sum and tf.reduce_mean are the same, since we will normalize the gradient (tf.sign) to update.

**Comment by qilong-zhang** [comment]
> Right. We actually normalize the gradient according to each channel. Note that tf.reduce_sum and tf.reduce_mean are the same, since we will normalize the gradient (tf.sign) to update.

But before you do tf.sign, the code will first update noise
```python
noise = momentum * grad + noise
```
Maybe after this operation, the result of tf.sign will be different from that we use tf.reduce_sum?

**Comment by dongyp13** [comment]
Why?  'grad' is also a summation of the previous gradients.

**Comment by qilong-zhang** [comment]
Oh！I see， thanks！
